contrib/gitian-builder/setup/dependencies.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+# Copyright (c) 2021 The Dogecoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+oss_patch_url="https://bitcoincore.org/cfields/osslsigncode-Backports-to-1.7.1.patch"
+oss_patch_hash="a8c4e9cafba922f89de0df1f2152e7be286aba73f78505169bc351a7938dd911"
+
+oss_tar_url="https://launchpad.net/ubuntu/+archive/primary/+sourcefiles/osslsigncode/1.7.1-1/osslsigncode_1.7.1.orig.tar.gz"
+oss_tar_hash="f9a8cdb38b9c309326764ebc937cba1523a3a751a7ab05df3ecc99d18ae466c9"
+
+macos_sdk_url="https://bitcoincore.org/depends-sources/sdks/MacOSX10.11.sdk.tar.gz"
+macos_sdk_hash="bec9d089ebf2e2dd59b1a811a38ec78ebd5da18cbbcd6ab39d1e59f64ac5033f"
+
+mkdir -p inputs
+
+function check_file() {
```

Comment:
Not sure about the invalid file. There is no default value here

```
if [[ $? == 0 ]]; then
    echo "OK"
  else
    echo ""
  fi
```

Reply:
Just by changing url of `oss_patch_url` for example, if you download a totally different file, and in worst case the same filename with some malicious content.

You will download your file, check your shasum using `echo "${1} $(basename -- $2)" | sha256sum -c > /dev/null 2>&1`.
If it's invalid, you will do your condition, but the issue is in case of an invalid file, you should exit on the following `check_file "$2" "$1" || { echo "Signature for $(basename -- $1) don't match"; exit 1; }`.
Because your `check_file` condition do an echo in any case, his return will be always 0, and will never enter the OR condition who display "Signature don't match" and exit.

You will continue the script execution, and may compile an untrusted source :).

Reply:
Fixed!